src/lib.rs

```diff
@@ -148,25 +148,25 @@ fn get_literal_str(l: Lit) -> String {
     }
 }
 
+/// Check if a return type is some form of `Result`. This assumes that all types named `Result`
+/// are in fact results, but is resilient to the possibility of `Result` types being referenced
+/// from specific modules.
+pub(crate) fn is_result_type(ty: &syn::TypePath) -> bool {
+    if let Some(segment) = ty.path.segments.iter().last() {
+        segment.ident == "Result"
+    } else {
+        false
+    }
+}
+
 fn check_if_return_result(f: &ItemFn) -> bool {
-    let retrn = &f.decl.output;
-    match retrn {
-        ReturnType::Default => false,
-        ReturnType::Type(_, t) => {
-            match  *t.clone() {
-                Type::Path(tp) => {
-                    if tp.path.segments.is_empty() {
-                        false
-                    } else {
-                        tp.path.segments[0].ident == "Result"
-                    }
-                },
-                _ => false,
-            }
+    if let ReturnType::Type(_, t) = &f.decl.output {
+        if let Type::Path(path) = t.as_ref() {
```

Comment:
Please keep this a match because in the future I might want to add support for other special cases(i.e tuple)

Reply:
Done